library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c

```diff
@@ -222,6 +252,10 @@ static libspdm_return_t libspdm_try_negotiate_algorithms(libspdm_context_t *spdm
         status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
         goto receive_done;
     }
+    if (spdm_response->header.param1 > req_param1) {
```

Comment:
I think we have discussed this in previous PR.
The spec is not clear about responder's behavior, so I recommend to loosen the check.

Reply:
I've dropped this check on the requester as it seems too controversial for the time being :)